### PR TITLE
Joystick turbine interface

### PR DESCRIPTION
**Pull request by benkroop**
Replacement for https://github.com/build-week/hover-jet/pull/89. I accidentally merged that PR into a branch that had previously been merged into mater.

### CODE REVIEW

File: infrastructure/joystick/yaw_control.cc [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      break;
+    }
+  }
+
+  while (true) {
+    auto pedal_event = pedals->read_event();
+    if (pedal_event) {
+      if (pedal_event->event_type == jet::EventType::AXIS) {
+        if (pedal_event->axis_id == YAW_AXIS_ID) {
+          double yaw = ((double)(*pedal_event->axis_value)) / (INT16_MAX);
```
Comment (jpanikulam) [inline infrastructure/joystick/yaw_control.cc]: Prefer static_cast

Comment (jpanikulam) [reply infrastructure/joystick/yaw_control.cc]: Is it actually on the range from 0-> int16 max?


Also: Should consider doing `std::numeric_limits<uint16_t>::max()` vs the C macro (I assume that's the C macro)

File: infrastructure/joystick/joystick_test.cc [hunk 2]
```diff
@@ -2,7 +2,8 @@
 #include "infrastructure/joystick/joystick.hh"
 
 int main(int argc, char *argv[]) {
-  jet::Joystick joystick("/dev/input/js0");
+  jet::Joystick joystick("/dev/input/js2");
+  // jet::Joystick joystick("/dev/input/by-id/usb-Thustmaster_Joystick_-_HOTAS_Warthog-joystick");
```
Comment (jpanikulam) [inline infrastructure/joystick/joystick_test.cc]: Dead code

File: infrastructure/joystick/yaw_control.cc (hunk 1 above)
Comment (benkroop) [reply infrastructure/joystick/yaw_control.cc]: Yep, should update to use static cast and numeric_limits. And the range is actually exactly 0 -> INT16MAX.

**Comment by benkroop** [comment]
Updated PR with your requested changes. Will merge as soon as CI passes.

**Pull request merged**